sike-java/src/main/java/com/wultra/security/pqc/sike/model/EncryptedMessage.java

```
@@ -73,7 +73,7 @@ public EncryptedMessage(SikeParam sikeParam, byte[] bytes) {
      */
     public byte[] getEncoded() {
         if (c0 == null || c1 == null) {
-            return null;
+            return new byte[0];
```

@romanstrobl Please review the semantics here, maybe we throw an error somewhere in case of null?

---

Although this change does not impact current code, I do not like the change, because a comparison function which compares two messages encoded using `getEncoded()` could consider them as equal if both where `new byte[0]`. I prefer to stick with the `null` value for this reason.

---

I was paying attention to [MSC19-C](https://wiki.sei.cmu.edu/confluence/display/c/MSC19-C.+For+functions+that+return+an+array%2C+prefer+returning+an+empty+array+over+a+null+value) when I made the change. In this codebase, returning null is not an issue, but in the future it will force callers of getEncoded() to test for nullity, making code less readable and maybe more complex (that was at least my thought).

---

I would stick with @romanstrobl on this one and use `null` as a return value. The semantics of Java cryptographic primitives uses `null` as an acceptable return value for data encoding, see for example:

https://docs.oracle.com/en/java/javase/15/docs/api/java.base/java/security/Key.html#getEncoded()

The consuming code usually needs to handle `null` values anyway. For example, the implementation of `X509Key.getEncoded()` (the base class for many practical keys) looks like this:

```java
    public byte[] getEncoded() {
        try {
            return (byte[])this.getEncodedInternal().clone();
        } catch (InvalidKeyException var2) {
            return null;
        }
    }

    public byte[] getEncodedInternal() throws InvalidKeyException {
        byte[] var1 = this.encodedKey;
        if (var1 == null) {
            try {
                DerOutputStream var2 = new DerOutputStream();
                this.encode(var2);
                var1 = var2.toByteArray();
            } catch (IOException var3) {
                throw new InvalidKeyException("IOException : " + var3.getMessage());
            }

            this.encodedKey = var1;
        }

        return var1;
    }
```

---

Changed on 7b01fa3fd7ae38575e3917d7b4addad39f255ed3